Handle nans via `is_valid_value()` (#2613)
geopm/geopm: service/docs/source/geopm_pio.3.rst

       int geopm_pio_reset(void);

       int geopm_pio_check_valid_value(double value);

> 

Reviewer: ```suggestion

```

  Returns zero on success, error value on failure.

``geopm_pio_reset()``
  Reset the GEOPM platform interface causing resources to be freed.
  This will cause the internal PlatormIO instance to be
  released/deleted and reconstructed.  As a result, any signals and
  controls that had been pushed will be cleared, any batch servers
  that had been started will be stopped, and all registered IOGroups
  will be reset.  **NOTE: the reset only applies to the Service
  PlatformIO instance and does not affect the PlatformIO instance

>   managed by the GEOPM HPC runtime.**

Reviewer: This comment is completely unrelated to this pull request, but this Note is misleading.  The reset applies to all use cases of PlatformIO with the one exception of the GEOPM HPC Runtime use case.  In the GEOPM HPC Runtime usecase, a different singleton access function is used, so this reset does not apply to the HPC Runtime PlatformIO instance.

``geopm_pio_check_valid_value()``

>   In geopm, the values for signals and controls can be invalid due to errors.

Reviewer: ```suggestion
  In geopm, signals may be unavailable due to missing information.  In these cases
  where the requested signal temporarily unavailable an invalid value is returned.
 ```

>   This function determines if the current *value* of the signal or control is

Reviewer: ```suggestion
  This function determines if the current *value* of the signal is
```
